### PR TITLE
parameterize asset graph tests

### PR DESCRIPTION
**Pull request by johannkm**
Clean up the test suite with a pytest fixture for internal and external asset graphs

### CODE REVIEW

**Comment by johannkm** [comment]
Current dependencies on/for this PR:
* master
  * **PR #16819** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/16819" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/16819?utm_source=stack-comment).

File: python_modules/dagster/dagster_tests/asset_defs_tests/test_asset_graph.py [hunk 1]
```diff
@@ -385,13 +334,16 @@ def test_partitioned_source_asset():
     def downstream_of_partitioned_source():
         pass
 
-    asset_graph = AssetGraph.from_assets([partitioned_source, downstream_of_partitioned_source])
+    asset_graph = asset_graph_from_assets([partitioned_source, downstream_of_partitioned_source])
+
+    if isinstance(asset_graph, ExternalAssetGraph):
+        pytest.xfail("not supported with ExternalAssetGraph")
```
Comment (johannkm) [inline python_modules/dagster/dagster_tests/asset_defs_tests/test_asset_graph.py:340]: I didn't dig in. `assert asset_graph.is_partitioned(AssetKey("partitioned_source"))` is False with an external graph

File: python_modules/dagster/dagster_tests/asset_defs_tests/test_asset_graph.py [hunk 2]
```diff
@@ -192,35 +163,26 @@ def child(parent): ...
...
-
-            mapped_partitions_result = external_asset_graph.get_parents_partitions(
+            mapped_partitions_result = asset_graph.get_parents_partitions(
                 instance, current_time, child.key
             )
-            mapped_partitions_result.parent_partitions == {
+            assert mapped_partitions_result.parent_partitions == {
                 AssetKeyPartitionKey(parent.key, "2022-01-02")
             }
-            mapped_partitions_result.required_but_nonexistent_parents_partitions == {}
```
Comment (johannkm) [inline python_modules/dagster/dagster_tests/asset_defs_tests/test_asset_graph.py:223]: missing asserts, we should lint for this

File: python_modules/dagster/dagster_tests/asset_defs_tests/test_asset_graph.py (hunk 1 above)
Comment (johannkm) [reply python_modules/dagster/dagster_tests/asset_defs_tests/test_asset_graph.py:340]: (previously this test was only on internal)

**Review by clairelin135 — APPROVED** [review]
This looks good to me -- thanks for the refactor!

**Pull request merged**